genie-web/src/main/java/com/netflix/genie/web/tasks/leader/AgentJobCleanupTask.java

```diff
@@ -0,0 +1,145 @@
+/*
+ *
+ *  Copyright 2019 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.tasks.leader;
+
+import com.google.common.collect.Maps;
+import com.netflix.genie.common.dto.JobStatus;
+import com.netflix.genie.common.exceptions.GenieException;
+import com.netflix.genie.web.properties.AgentCleanupProperties;
+import com.netflix.genie.web.services.JobPersistenceService;
+import com.netflix.genie.web.services.JobSearchService;
+import com.netflix.genie.web.tasks.GenieTaskScheduleType;
+import lombok.extern.slf4j.Slf4j;
+
+import java.time.Instant;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Leader task that cleans up jobs whose agent crashed or disconnected.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+@Slf4j
+public class AgentJobCleanupTask extends LeadershipTask {
```

Comment:
It would be nice if you published some metrics on this. It doesn't have to be part of this PR but similar to how we track "lost jobs" this would be good information to have.

Reply:
Yes i was gonna, but wanted to make sure this goes in today so left it in a (WIP) commmit.
I'm adding 
 - Gauge with currently disconnected jobs (size of the map)
 - Counter of jobs marked killed
Anything else you can think of?

Reply:
Counters may be kind of screwy as the interval is so long apart we'll just see random bursts. I think I used a gauge for that with lost jobs then each value reported stayed until the next iteration.

Reply:
They're not random bursts.. we increment the count when a job is killed.
So yes, it is bursty but it's what's actually happening.

Reply:
Meaning there is no information we can really glean from a counter. It'll just be spikes at some rate. Counters are only valuable when there is changes in rates over time. This will go from 0 to some number and we'll just end up normalizing it to the raw number anyway by dividing by the duration to get a value. So to me the count serves no real purpose but we can get the same info from a counter as a gauge we'll just have to find the spikes.

Reply:
I don't get it. We're just counting the number of events (jobs marked failed in this case).

What's wrong with this approach?